### PR TITLE
add script to automate inference on directory files

### PR DESCRIPTION
**Pull request by jsedillo**

### CODE REVIEW

**Review by yzamd — APPROVED** [review]

File: loop.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      dep_2 = os.path.join(dep_dir, src_deps[int(i / interval) + 1])
+      cmd += f" --depOne {dep_1} --depTwo {dep_2}"
+
+    # Generate output file name
+    i_interp = (2 * i + interval) / 2
+    file_out = f"{i_interp:08.1f}" if i_interp % 1 else f"{int(i_interp):06d}"
+    file_out = f"{out_dir}/{file_out}_interp.png"
+    cmd += f" --out {file_out}"
+
+    print(cmd)
```
Comment (fanghAMD) [inline loop.py:85]: no need to print here.

File: loop.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    os.system(cmd)
```
Comment (fanghAMD) [inline loop.py:86]: i am thinking could it be better to generate a list of cmds and append in a list.

then call cmd. this way, its easier to debug script.

File: loop.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import os
+import argparse
+
+BASE_CMD = "python main.py --model lf"
+SOURCE_DIR = "./Insomniac_Spiderman1_Interaction_Inidcator_02"
+OUTPUT_DIR = "./output"
+SKIP_INTERVAL = 2
+
+
+def loop(input=SOURCE_DIR, output=OUTPUT_DIR, interval=SKIP_INTERVAL, vel=False, dep=False):
```
Comment (fanghAMD) [inline loop.py]: function should not be called loop.
too general

File: loop.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import os
+import argparse
+
+BASE_CMD = "python main.py --model lf"
+SOURCE_DIR = "./Insomniac_Spiderman1_Interaction_Inidcator_02"
```
Comment (fanghAMD) [inline loop.py]: better to take cmd arguments and have these default settings in launch.json

File: loop.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    out_dir += "Depth"
+  if not vel and not dep:
+    out_dir += "None"
+  os.makedirs(out_dir, exist_ok=True)
+
+  # Load list of files
+  img_dir = input + "/col"
+  src_imgs = os.listdir(img_dir)
+  if vel:
+    vel_dir = input + "/vel"
```
Comment (fanghAMD) [inline loop.py]: add comments to show the required folder hierachy. 

easier for maintenance.

File: loop.py (hunk 1 above)
Comment (jsedillo) [reply loop.py:85]: I added it to give debugging information when errors occur. I consider it a good alternative compared to executing a list of commands like you suggested. Let me know which one you prefer.

File: loop.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    cmd += f" --out {file_out}"
+
+    print(cmd)
+    os.system(cmd)
+
+
+if __name__ == "__main__":
+  parser = argparse.ArgumentParser(description="Loop through directory option")
+  parser.add_argument("-i", "--input", help="Input directory with all source data.", type=str)
+  parser.add_argument("-0", "--output", help="Output directory with all source data.", type=str)
```
Comment (jsedillo) [inline loop.py]: Just noticed this is a 0 instead of o for some reason. Will fix.

**Pull request merged**